dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorRecordStore.cs

```diff
@@ -0,0 +1,399 @@
+        Verify.NotNullOrWhiteSpace(key);
+
+        // Create Options
+        var collectionName = this.ChooseCollectionName(options?.CollectionName);
+        var maybePrefixedKey = this.PrefixKeyIfNeeded(key, collectionName);
+
+        // Get the redis value.
+        var redisResult = await RunOperationAsync(
+            collectionName,
+            "GET",
+            () => options?.IncludeVectors is true ?
+                this._database
+                    .JSON()
+                    .GetAsync(maybePrefixedKey) :
+                this._database
+                    .JSON()
+                    .GetAsync(maybePrefixedKey, this._dataPropertyNames)).ConfigureAwait(false);
+
+        // Check if the key was found before trying to parse the result.
+        if (redisResult.IsNull || redisResult is null)
+        {
+            throw new VectorStoreOperationException($"Could not find document with key '{key}'");
+        }
+
+        // Check if the value contained any json text before trying to parse the result.
+        var redisResultString = redisResult.ToString();
+        if (redisResultString is null)
+        {
+            throw new VectorStoreRecordMappingException($"Document with key '{key}' does not contain any json.");
+        }
+
+        // Convert to the caller's data model.
+        return RunModelConversion(
+            collectionName,
+            "GET",
+            () =>
+            {
+                var node = JsonSerializer.Deserialize<JsonNode>(redisResultString)!;
+                return this._mapper.MapFromStorageToDataModel((key, node));
+            });
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<TRecord> GetBatchAsync(IEnumerable<string> keys, GetRecordOptions? options = default, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(keys);
+        var keysList = keys.ToList();
+
+        // Create Options
+        var collectionName = this.ChooseCollectionName(options?.CollectionName);
+        var maybePrefixedKeys = keysList.Select(key => this.PrefixKeyIfNeeded(key, collectionName));
+        var redisKeys = maybePrefixedKeys.Select(x => new RedisKey(x)).ToArray();
+
+        // Get the list of redis results.
+        var redisResults = await RunOperationAsync(
+            collectionName,
+            "MGET",
+            () => this._database
+                .JSON()
+                .MGetAsync(redisKeys, "$")).ConfigureAwait(false);
+
+        // Loop through each key and result and convert to the caller's data model.
+        for (int i = 0; i < keysList.Count; i++)
+        {
+            var key = keysList[i];
+            var redisResult = redisResults[i];
+
+            // Check if the key was found before trying to parse the result.
+            if (redisResult.IsNull || redisResult is null)
+            {
+                throw new VectorStoreOperationException($"Could not find document with key '{key}'");
+            }
```

Comment:
If I'm getting a batch and a record is deleted then the entire operation fails and I have to go again. Is this correct and is this the behaviour we want?

Reply:
Agreed.  I haven't tested all the different SDKs we are evaluating though, since setting each one up, would take quite a while.  In terms of per item error results (and here I'm assuming we are referring to errors other than not found), I agree that it's useful, and makes for a higher quality API, but as part of my research I found a few of the DBs we are looking at doesn't support this.  At best we can do this for not found.
The good thing is that all the batch operations we support are idempotent, so retrying the entire thing isn't a problem.

Reply:
I've implemented and pushed option 1 from above.